### PR TITLE
Add new rule to transform package metadata for FUS

### PR DESCRIPTION
**Pull request by smukherj1**
The input format is a CSV with package names & versions as generated by
download_pkgs in base_images_docker. The output format is a YAML as
supported by FUS. FUS requires the metadata YAML to be a dictionary.

### CODE REVIEW

File: rules/container/packages_metadata.bzl [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+def _impl(ctx):
+    ctx.actions.run(
+        inputs = [ctx.file.metadata_csv],
+        outputs = [ctx.outputs.yaml],
+        executable = ctx.executable._converter,
+        arguments = ["-inputCSV", ctx.file.metadata_csv.path, "-outputYAML", ctx.outputs.yaml.path],
+        mnemonic = "PackagesMetadataYAML",
+    )
+    return
```
Comment (nlopezgi) [inline rules/container/packages_metadata.bzl]: do you need an explicit return?

File: rules/container/packages_metadata.bzl [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+def _impl(ctx):
+    ctx.actions.run(
+        inputs = [ctx.file.metadata_csv],
+        outputs = [ctx.outputs.yaml],
+        executable = ctx.executable._converter,
+        arguments = ["-inputCSV", ctx.file.metadata_csv.path, "-outputYAML", ctx.outputs.yaml.path],
```
Comment (nlopezgi) [inline rules/container/packages_metadata.bzl]: wrap around: put one arg per line

File: tools/go/cmd/packages_metadata/packages_metadata.go [hunk 3]
```diff
@@ -0,0 +1,115 @@
+// This binary accepts a CSV file with columns:
```
Comment (nlopezgi) [inline tools/go/cmd/packages_metadata/packages_metadata.go]: move to src/go/cmd/...

File: tools/go/cmd/packages_metadata/packages_metadata.go [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+	sort.Slice(result.Packages, func(i, j int) bool {
+		pi, pj := result.Packages[i], result.Packages[j]
+		return strings.Compare(pi.Name, pj.Name) < 0
+	})
+	return result, nil
+}
+
+func main() {
+	flag.Parse()
+	log.Println("Running the download_pkgs metadata CSV to YAML converter.")
```
Comment (nlopezgi) [inline tools/go/cmd/packages_metadata/packages_metadata.go]: remove reference to download_pkgs here?

File: rules/container/packages_metadata.bzl (hunk 1 above)
Comment (smukherj1) [reply rules/container/packages_metadata.bzl]: Done

File: rules/container/packages_metadata.bzl (hunk 2 above)
Comment (smukherj1) [reply rules/container/packages_metadata.bzl]: Done

File: tools/go/cmd/packages_metadata/packages_metadata.go (hunk 3 above)
Comment (smukherj1) [reply tools/go/cmd/packages_metadata/packages_metadata.go]: Done

File: tools/go/cmd/packages_metadata/packages_metadata.go (hunk 4 above)
Comment (smukherj1) [reply tools/go/cmd/packages_metadata/packages_metadata.go]: Done

File: rules/container/packages_metadata.bzl [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+packages:
+    - name: package1
+      version: version1
+    - name: package2
+      version: version2
+    ...
+"""
+packages_metadata = rule(
+    attrs = {
+        "metadata_csv": attr.label(allow_single_file = True),
```
Comment (nlopezgi) [inline rules/container/packages_metadata.bzl]: set mandatory?

Comment (smukherj1) [reply rules/container/packages_metadata.bzl]: Done

**Pull request merged**